pmd-core/src/main/java/net/sourceforge/pmd/util/ClasspathClassLoader.java

```diff
@@ -57,17 +58,19 @@ public ClasspathClassLoader(String classpath, ClassLoader parent) throws IOExcep
         return urlList.toArray(new URL[0]);
     }
 
-    private static URL[] initURLs(String classpath) throws IOException {
-        if (classpath == null) {
-            throw new IllegalArgumentException("classpath argument cannot be null");
-        }
+    private static URL[] initURLs(String classpath) {
+        AssertionUtil.requireParamNotNull("classpath", classpath);
         final List<URL> urls = new ArrayList<>();
-        if (classpath.startsWith("file:")) {
-            // Treat as file URL
-            addFileURLs(urls, new URL(classpath));
-        } else {
-            // Treat as classpath
-            addClasspathURLs(urls, classpath);
+        try {
+            if (classpath.startsWith("file:")) {
+                // Treat as file URL
+                addFileURLs(urls, new URL(classpath));
+            } else {
+                // Treat as classpath
+                addClasspathURLs(urls, classpath);
+            }
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Cannot prepend classpath " + classpath + "\n" + e.getMessage(), e);
```

Comment:
I changed back PMDConfiguration, now there is no source incompatibility

Reply:
Actually, now we don't throw IOException anymore - it will always be a IllegalArgumentException. If old code uses "prependClasspath" it has it's error handling code on IOException. I would keep the behavior and change the exception only, when "prependAuxclasspath" is used.